tests/auto/qhttpserver/tst_qhttpserver.cpp

```diff
@@ -349,13 +349,18 @@ void tst_QHttpServer::initTestCase()
             [expectedSslErrors](QNetworkReply *reply,
                                 const QList<QSslError> &errors) {
         for (const auto &error: errors) {
+            bool expected{false};
             for (const auto &expectedError: expectedSslErrors) {
-                if (error.error() != expectedError.error() ||
-                    error.certificate() != expectedError.certificate()) {
-                    qCritical() << "Got unexpected ssl error:"
-                                << error << error.certificate();
+                if (error.error() == expectedError.error() ||
+                    error.certificate() == expectedError.certificate()) {
+                    expected = true;
```

Comment:
I happened to come across this pull request, and I wondered if the `or` in the logic should be changed into an `and`, since we're inverting the logic in the original code. Both there error type *and* the associated error certificate in the expected should match the actual error.